src/agent/codecoveragesummaryreader.ts

```diff
@@ -0,0 +1,219 @@
+import ifm = require('./interfaces');
+import testifm = require('vso-node-api/interfaces/TestInterfaces');
+import ccp = require('./codecoveragepublisher');
+import utilities = require('./utilities');
+import cm = require('./common');
+import Q = require('q');
+
+var xmlreader = require('xmlreader');
+
+export class CodeCoverageSummary {
+    results: testifm.CodeCoverageStatistics[];
+
+    constructor() {
+        this.results = [];
+    }
+
+    addResults(res) {
+        this.results = this.results.concat(res);
+    }
+}
+
+export class JacocoSummaryReader implements ccp.ICodeCoverageReader {
+
+    private command: cm.ITaskCommand;
+
+    constructor(command: cm.ITaskCommand) {
+        this.command = command;
+    }
+
+    public getCodeCoverageSummary(summaryFilePath: string): Q.Promise<testifm.CodeCoverageData> {
+        var defer = Q.defer<testifm.CodeCoverageData>();
+        var _this = this;
+
+        SummaryReaderUtilities.getXmlContent(summaryFilePath).then(function(xmlContents) {
```

Comment:
Already checking if xmlcontent has report nodes .will also add a null check for xmlcontent